index.js

```diff
@@ -125,16 +193,42 @@ module.exports = () => {
      * added / dropped peers.
      */
     _sendMessage () {
-      var localAdded = Object.keys(this._localAddedPeers).slice(0, PEX_MAX_PEERS)
-      var localDropped = Object.keys(this._localDroppedPeers).slice(0, PEX_MAX_PEERS)
-
-      var added = Buffer.concat(localAdded.map(string2compact))
-      var dropped = Buffer.concat(localDropped.map(string2compact))
-
-      var addedFlags = Buffer.concat(localAdded.map(() => {
-        // TODO: support flags
-        return Buffer.from([0])
-      }))
+      const localAdded = Object.keys(this._localAddedPeers).slice(0, PEX_MAX_PEERS)
+      const localDropped = Object.keys(this._localDroppedPeers).slice(0, PEX_MAX_PEERS)
+
+      const _isIPv4 = (peers, addr) => peers[addr].ip === 4
+      const _isIPv6 = (peers, addr) => peers[addr].ip === 6
+      const _flags = (peers, addr) => peers[addr].flags
+
+      const added = localAdded
+        .filter(k => _isIPv4(this._localAddedPeers, k))
+        .map(string2compact)
+        .reduce((acc, cur) => Buffer.concat([acc, cur]), Buffer.alloc(0))
+
+      const added6 = localAdded
+        .filter(k => _isIPv6(this._localAddedPeers, k))
+        .map(string2compact)
+        .reduce((acc, cur) => Buffer.concat([acc, cur]), Buffer.alloc(0))
+
+      const dropped = localDropped
+        .filter(k => _isIPv4(this._localDroppedPeers, k))
+        .map(string2compact)
+        .reduce((acc, cur) => Buffer.concat([acc, cur]), Buffer.alloc(0))
+
+      const dropped6 = localDropped
+        .filter(k => _isIPv6(this._localDroppedPeers, k))
+        .map(string2compact)
+        .reduce((acc, cur) => Buffer.concat([acc, cur]), Buffer.alloc(0))
+
+      const addedFlags = localAdded
+        .filter(k => _isIPv4(this._localAddedPeers, k))
+        .map(k => _flags(this._localAddedPeers, k))
+        .reduce((acc, cur) => Buffer.from([cur]), Buffer.alloc(0))
```

Comment:
should this be 

```js
.map(flagInt => Buffer.from([flagInt]))
.reduce((acc, cur) => Buffer.concat([acc, cur]), Buffer.alloc(0))
```

? (and for `add6Flags` too)

Reply:
Nice catch! This was actually a bug, so I fixed and added more test 👍